rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -225,10 +225,20 @@ where
                 .await?
                 .saturating_add(U256::from(GAS_ESTIMATE_BUFFER))
         };
+        let chain_id = match self.provider.get_chainid().await {
+            Ok(chainId) => chainId.as_u32(),
+            // Couldn't get chainId, assume not 1559
+            Err(_) => return Ok(tx.gas(gas_limit))
+        };
         let Ok((max_fee, max_priority_fee)) = self.provider.estimate_eip1559_fees(None).await else {
             // Is not EIP 1559 chain
             return Ok(tx.gas(gas_limit))
         };
+        let max_priority_fee = if KnownHyperlaneDomain::try_from(chain_id)? == KnownHyperlaneDomain::Polygon {
```

Comment:
Maybe use
```rust
if matches!(KnownHyperlaneDomain::try_from(chain_id), Ok(KnownHyperlaneDomain::Polygon)) {..} else {..}
```

Reply:
Would you mind sharing what the benefit of that is? It seems less readable to me?

Reply:
We shouldn't use `?` because then an unknown chain (i.e. a PI chain) that supports the `eth_chainId` RPC will always have errors

My suggestion is to get rid of the `let chain_id` above altogether, which imo isn't needed and is potentially inaccurate because we have access to `message.destination`. The EVM chain_id isn't always going to be the Hyperlane domain ID-- it often is, but it's not guaranteed to be, especially not in PI chains

And instead I think we should do exactly what mattie suggests, which involves the `else` block being used if the message.destination isn't a "known" domain. But instead of chain_id use message.destination:
```
if matches!(KnownHyperlaneDomain::try_from(message.domain), Ok(KnownHyperlaneDomain::Polygon)) {..} else {..}
```

Reply:
Ahhhh now I get it